text/0000-else.md

```diff
@@ -0,0 +1,293 @@
+<template lwc:else></template>
+```
+
+```html
+<!-- Invalid -->
+<template lwc:if={foo}></template>
+<template lwc:else={yolo}></template>
+```
+
+Similarly, `lwc:if` and `lwc:elseif` _must_ have an expression as their value:
+
+```html
+<!-- Valid -->
+<template lwc:if={foo}></template>
+<template lwc:elseif={bar}></template>
+```
+
+```html
+<!-- Invalid -->
+<template lwc:if="foo"></template>
+<template lwc:elseif="bar"></template>
+```
+
+```html
+<!-- Invalid -->
+<template lwc:if={foo}></template>
+<template lwc:elseif></template>
+```
+
+### Semantics
+
+`lwc:if` behaves the same as `if:true`, except that it also supports an initial `!` in the expression.
+
+`lwc:elseif` behaves the same as `lwc:if`, except that its behavior depends on any preceding `lwc:if`/`lwc:elseif` statements.
+
+Similarly, the behavior of `lwc:else` depends on its previous siblings.
+
+For example, in a series of sibling `lwc:if` / `lwc:elseif` / `lwc:else` directives, the inverse of the previous evaluated expression determines whether we reach the next directive:
+
+```html
+<template lwc:if={abra}>
+    Abra!
+</template>
+<template lwc:elseif={kadabra}>
+    Kadabra!
+</template>
+<template lwc:elseif={hocus}>
+    Hocus pocus!
+</template>
+<template lwc:else>
+    Kadabra!
+</template>
+```
+
+Rather than rehash the basics of conditional logic in programming, let's define this as being analogous to the [`if` statement](https://262.ecma-international.org/12.0/#prod-IfStatement) (and friends) in JavaScript:
+
+```js
+if (component.abra) {
+  // Abra!
+} else if (component.kadabra) {
+  // Kadabra!
+} else if (component.hocus) {
+  // Hocus pocus!
+} else {
+  // Kadabra!
+}
+```
+
+Note that this means:
+
+1. The property getters are only accessed once per instance of an `lwc:if` or `lwc:ifelse`.
+2. Property getter access is determined by the ordering. In the above case, if `component.abra` is truthy, then none of the other property getters will be accessed.
+
+## Drawbacks
+
+This implementation would require the template parser to have knowledge of sibling nodes, rather than treating each sibling node independently.
+
+This implementation would also supersede the existing `if:true` and `if:false` directives, which may lead to some confusion for developers.
+
+## Alternatives
+
+The impact of _not_ doing this is that it's much trickier to do conditional logic in LWC templates.
+
+## Adoption strategy
+
+This is a non-breaking change, so users can opt-in as they see fit.
+
+# How we teach this
+
+This feature is very similar to the same concept in other frameworks (Vue, Svelte, Angular, etc.). It's also similar to `if` and `else` in JavaScript. So we can leverage that familiarity when teaching this.
+
+As for the existing `if:true` and `if:else` statements, we will need to mark them as "legacy," since they are rendered redundant by the new `lwc:if`. To ease the transition, we can provide:
+
+1. ESLint rules to warn about using `if:true` and `if:else`.

```

Comment:
```suggestion
1. Make the template compiler warn about using `if:true` and `if:else`.
```

ESLint doesn't run against template code. I would suggest adding making the template compiler warn. We can create an HTML parser for ESLint but it would require a lot more work.